http/src/test/java/com/redhat/lightblue/client/http/transport/JavaNetHttpClientIntegrationTest.java

```diff
@@ -0,0 +1,67 @@
+package com.redhat.lightblue.client.http.transport;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlMatching;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+import com.redhat.lightblue.client.http.HttpMethod;
+import com.redhat.lightblue.client.http.testing.doubles.FakeLightblueRequest;
+import com.redhat.lightblue.client.request.LightblueRequest;
+
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.IOException;
+
+@RunWith(JUnit4.class)
+public class JavaNetHttpClientIntegrationTest {
+    @Rule
+    public WireMockRule wireMockRule = new WireMockRule();
```

Comment:
Wiremock [very quickly] starts an http server for the tests. This is to test the real semantics of java.net HttpUrlConnection which I wasn't totally familiar with (how error responses are dealt with in particular).